### ISSUE TITLE
SDL backend: DragFloat exploding and freezing

### ISSUE THREAD

**Issue by exsec** [ocornut/imgui#1559]
![image](https://user-images.githubusercontent.com/24857556/35136553-62846e08-fccb-11e7-9c46-62f86dd0fec1.png)
Hi, I'm having a problem when I drag the mouse off the window when using a drag float widget, I'm using the SDL2 example.
When out of window the float goes to negative huge number and gives nan after that.

**Comment by ocornut** [comment]
Hello,

There are two sides to this bug:

1. The SDL backend currently doesn't report valid mouse coordinates outside the window boundaries.
I have made a first quick change to the SDL backends to use `SDL_WINDOW_INPUT_FOCUS` instead of `SDL_WINDOW_MOUSE_FOCUS` which will report clamped mouse coordinates (not ideal, but better). 

To be correct we should be using `SDL_CaptureMouse()` + test the `SDL_WINDOW_MOUSE_CAPTURE` flag, both were added in SDL 2.0.4 (released 2016-01-02) which is a little recent-ish in SDL terms. I will investigate that now.

2. The DragFloat() is expecting the backend to report valid mouse coordinates when active.
I have made a change to DragFloat to handle that case now.
